<launch>

    <arg name="do_slam" default="true"/>
    <arg name="vehicle_name" default="MyVehicle"/>
    <arg name="vehicle_frame" default="base_link"/>
    <arg name="odom_frame" default="odom"/>
    <arg name="map_frame" default="map"/>
    <arg name="world_frame" default="world"/>
    <arg name="image_pub_freq" default=".05"/>
    <arg name="lidar_pub_freq" default=".1"/>
    <arg name="control_pub_freq" default=".1"/>
    <arg name="update_imu_n_sec" default=".1"/>
    <arg name="enable_cameras" default="false"/>
    <arg name="raw_lidar_topic" value="/airsim_ros_node/$(arg vehicle_name)/lidar/Lidar2"/>

    <node pkg="airsim_launch" type="airsim_node" respawn="false" name="airsim_ros_node" output="screen">
        <param name="update_airsim_img_response_every_n_sec" value="$(arg image_pub_freq)" />
        <param name="update_lidar_every_n_sec" value="$(arg lidar_pub_freq)" />
        <param name="update_airsim_control_every_n_sec" value="$(arg control_pub_freq)" />
        <param name="update_imu_n_sec" value="$(arg update_imu_n_sec)" />
        <param name="vehicle_frame_id" value="$(arg vehicle_frame)" />
        <param name="odom_frame_id" value="$(arg odom_frame)" />
        <param name="map_frame_id" value="$(arg map_frame)" />
        <param name="world_frame_id" value="$(arg world_frame)" />
        <param name="enable_cameras" value="$(arg enable_cameras)" />
    </node>

    <group if="$(arg do_slam)">
        <include file="$(find rs_to_velodyne)/launch/rs_to_velodyne.launch">
            <arg name="raw_lidar_topic" value="$(arg raw_lidar_topic)"/>
            <arg name="input_cloud_type" value="XYZI"/>
            <arg name="output_cloud_type" value="XYZIR"/>
        </include>

        <node pkg="tf" type="static_transform_publisher" name="velodyne_pub" args="0 0 0 0 0 0 base_link/Lidar2 velodyne 100"/>
    </group>

    <node type="rviz" name="airsim_rviz" pkg="rviz" args="-d $(find vehicle_launch)/config/airsim.rviz" output="log"/>
</launch>